### DISCUSSION TITLE
the same policy for reqid exists

### DISCUSSION THREAD

**Discussion by RusselWestbrook6** [strongswan/strongswan#1362] [Q&A]
This problem occurs when establishing an IPsec two-phase connection

Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 07[CFG] selected peer config "l2tp"
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 07[IKE] IKE_SA l2tp[41649] established between 172.16.31.48[172.16.31.48]...211.141.223.4[192.168.3.13]
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 07[IKE] scheduling reauthentication in 6625s
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 07[IKE] maximum IKE_SA lifetime 7165s
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 07[IKE] DPD not supported by peer, disabled
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 07[ENC] generating ID_PROT response 0 [ ID HASH ]
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 07[NET] sending packet: from 172.16.31.48[4500] to 211.141.223.4[4500] (76 bytes)
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 13[NET] received packet: from 211.141.223.4[4500] to 172.16.31.48[4500] (444 bytes)
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 13[ENC] parsed QUICK_MODE request 1 [ HASH SA No ID ID NAT-OA NAT-OA ]
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 13[CFG] selected proposal: ESP:AES_CBC_256/HMAC_SHA1_96/NO_EXT_SEQ
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 13[IKE] received 250000000 lifebytes, configured 0
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 13[ENC] generating QUICK_MODE response 1 [ HASH SA No ID ID NAT-OA NAT-OA ]
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 13[NET] sending packet: from 172.16.31.48[4500] to 211.141.223.4[4500] (204 bytes)
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[NET] received packet: from 211.141.223.4[4500] to 172.16.31.48[4500] (60 bytes)
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[ENC] parsed QUICK_MODE request 1 [ HASH ]
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[IKE] CHILD_SA l2tp{91993} established with SPIs c0b9938f_i 959b5b80_o and TS 172.16.31.48/32[udp/l2tp] === 211.141.223.4/32[udp/l2tp]
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown: 
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_VERSION: 1.1
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_VERB: up-host
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_CONNECTION: l2tp
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_INTERFACE: eth0
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_REQID: 29321
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_PROTO: esp
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_UNIQUEID: 41649
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_ME: 172.16.31.48
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_MY_ID: 172.16.31.48
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_MY_CLIENT: 172.16.31.48/32
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_MY_SOURCEIP: 
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_MY_PROTOCOL: 17
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_MY_PORT: 1701
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_PEER: 211.141.223.4
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_PEER_ID: 192.168.3.13
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_PEER_CLIENT: 211.141.223.4/32
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_PEER_SOURCEIP: 
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_PEER_PROTOCOL: 17
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_PEER_PORT: 1701
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_XAUTH_ID: 
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown:         PLUTO_UDP_ENC: 4500
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown: Cleanup: 222.71.171.98 47580 29308 down-host 10.254.135.41 1666141317
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown: Cleanup: 61.175.189.186 33958 27304 down-host 10.254.135.26 1666141432
Oct 19 09:07:12 13-59-shanghai-hw-cug kernel: [37101258.531807] <LWLog:Info:LWCONN>[LW_LinkUpdatePeerInfo:803] link {0:3294611} peer info changed, 0.0.0.0:49152 => 10.254.32.52:49152
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown: SNAT_IP_RANGE:  10.254.135.2-10.254.135.254, START: 10.254.135.2, END: 10.254.135.254
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown: iptables -t nat -I INPUT -p udp -s 211.141.223.4 --sport 4500 --dport 4500 -j SNAT --to 10.254.135.26
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown: iptables -t nat -I OUTPUT -p udp -d 211.141.223.4 --dport 4500 --sport 4500 -j DNAT --to 10.254.135.26
Oct 19 09:07:12 13-59-shanghai-hw-cug charon: 08[CHD] updown: conntrack -D --orig-src 211.141.223.4 -p udp --orig-port-src 4500
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 08[CHD] updown: conntrack v1.4.4 (conntrack-tools): 1 flow entries have been deleted.
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 08[CHD] updown: udp      17 179 src=211.141.223.4 dst=172.16.31.48 sport=4500 dport=4500 src=172.16.31.48 dst=211.141.223.4 sport=4500 dport=45
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 08[CHD] updown: 00 [ASSURED] mark=0 use=1
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 08[CHD] updown: conntrack -D --reply-dst 211.141.223.4 -p udp --reply-port-dst 4500
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 08[CHD] updown: conntrack v1.4.4 (conntrack-tools): 0 flow entries have been deleted.
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 08[CHD] updown: Connection 211.141.223.4 4500 29321 SNATed to 10.254.135.26
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 04[NET] received packet: from 211.141.223.4[4500] to 172.16.31.48[4500] (444 bytes)
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 04[ENC] parsed QUICK_MODE request 2 [ HASH SA No ID ID NAT-OA NAT-OA ]
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 04[CFG] selected proposal: ESP:AES_CBC_256/HMAC_SHA1_96/NO_EXT_SEQ
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 04[IKE] received 250000000 lifebytes, configured 0
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 04[IKE] detected rekeying of CHILD_SA l2tp{91993}
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 04[ENC] generating QUICK_MODE response 2 [ HASH SA No ID ID NAT-OA NAT-OA ]
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 04[NET] sending packet: from 172.16.31.48[4500] to 211.141.223.4[4500] (204 bytes)
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 10[IKE] retransmit 1 of request with message ID 328
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 10[NET] sending packet: from 172.16.31.48[4500] to 183.193.32.212[9249] (176 bytes)
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 11[NET] received packet: from 10.254.135.26[4500] to 172.16.31.48[4500] (444 bytes)
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 11[IKE] received retransmit of request with ID 2, retransmitting response
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 11[NET] sending packet: from 172.16.31.48[4500] to 211.141.223.4[4500] (204 bytes)
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 15[NET] received packet: from 183.193.32.212[9249] to 172.16.31.48[4500] (80 bytes)
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 15[ENC] parsed INFORMATIONAL response 328 [ ]
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 15[ENC] generating INFORMATIONAL request 329 [ N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) ]
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 15[NET] sending packet: from 172.16.31.48[4500] to 183.193.32.212[9249] (176 bytes)
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 07[NET] received packet: from 183.193.32.212[9249] to 172.16.31.48[4500] (80 bytes)
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 07[ENC] parsed INFORMATIONAL response 329 [ ]
Oct 19 09:07:13 13-59-shanghai-hw-cug charon: 06[IKE] sending keep alive to 60.174.148.25[1026]
Oct 19 09:07:14 13-59-shanghai-hw-cug charon: 08[NET] received packet: from 10.254.135.26[4500] to 172.16.31.48[4500] (444 bytes)
Oct 19 09:07:14 13-59-shanghai-hw-cug charon: 08[IKE] received retransmit of request with ID 2, retransmitting response
Oct 19 09:07:14 13-59-shanghai-hw-cug charon: 08[NET] sending packet: from 172.16.31.48[4500] to 211.141.223.4[4500] (204 bytes)
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 12[IKE] retransmit 1 of request with message ID 441
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 12[NET] sending packet: from 172.16.31.48[4500] to 39.144.153.162[60754] (176 bytes)
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 08[NET] received packet: from 39.144.153.162[60754] to 172.16.31.48[4500] (80 bytes)
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 08[ENC] parsed INFORMATIONAL response 441 [ ]
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 08[ENC] generating INFORMATIONAL request 442 [ N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) ]
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 08[NET] sending packet: from 172.16.31.48[4500] to 39.144.153.162[60754] (176 bytes)
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 04[NET] received packet: from 39.144.153.162[60754] to 172.16.31.48[4500] (80 bytes)
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 04[ENC] parsed INFORMATIONAL response 441 [ ]
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 04[IKE] received message ID 441, expected 442, ignored
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 05[NET] received packet: from 39.144.153.162[60754] to 172.16.31.48[4500] (144 bytes)
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 05[ENC] parsed INFORMATIONAL request 130 [ N(UPD_SA_ADDR) N(NATD_S_IP) N(NATD_D_IP) ]
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 05[ENC] generating INFORMATIONAL response 130 [ N(NATD_S_IP) N(NATD_D_IP) ]
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 05[NET] sending packet: from 172.16.31.48[4500] to 39.144.153.162[60754] (128 bytes)
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 09[NET] received packet: from 211.141.223.4[500] to 172.16.31.48[500] (408 bytes)
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 09[ENC] parsed ID_PROT request 0 [ SA V V V V V V V V ]
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 09[ENC] received unknown vendor ID: 01:52:8b:bb:c0:06:96:12:18:49:ab:9a:1c:5b:2a:51:00:00:00:01
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 09[IKE] received MS NT5 ISAKMPOAKLEY vendor ID
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 09[IKE] received NAT-T (RFC 3947) vendor ID
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 09[IKE] received draft-ietf-ipsec-nat-t-ike-02\n vendor ID
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 09[IKE] received FRAGMENTATION vendor ID
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 09[ENC] received unknown vendor ID: fb:1d:e3:cd:f3:41:b7:ea:16:b7:e5:be:08:55:f1:20
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 09[ENC] received unknown vendor ID: 26:24:4d:38:ed:db:61:b3:17:2a:36:e3:d0:cf:b8:19
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 09[ENC] received unknown vendor ID: e3:a5:96:6a:76:37:9f:e7:07:22:82:31:e5:ce:86:52
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 09[IKE] 211.141.223.4 is initiating a Main Mode IKE_SA
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 09[CFG] selected proposal: IKE:AES_CBC_256/HMAC_SHA1_96/PRF_HMAC_SHA1/MODP_2048
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 09[ENC] generating ID_PROT response 0 [ SA V V V V ]
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 09[NET] sending packet: from 172.16.31.48[500] to 211.141.223.4[500] (160 bytes)
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[NET] received packet: from 10.254.135.26[4500] to 172.16.31.48[4500] (76 bytes)
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[ENC] parsed INFORMATIONAL_V1 request 2912128556 [ HASH D ]
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CFG] unable to install policy 10.254.135.26/32[udp/l2tp] === 172.16.31.48/32[udp/l2tp] in for reqid 29321, the same policy for reqid 29228 exists
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CFG] unable to install policy 172.16.31.48/32[udp/l2tp] === 10.254.135.26/32[udp/l2tp] out for reqid 29321, the same policy for reqid 29228 exists
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[IKE] received DELETE for ESP CHILD_SA with SPI 959b5b80
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[IKE] closing CHILD_SA l2tp{91993} with SPIs c0b9938f_i (0 bytes) 959b5b80_o (0 bytes) and TS 172.16.31.48/32[udp/l2tp] === 10.254.135.26/32[udp/l2tp]
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown: 
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_VERSION: 1.1
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_VERB: down-host
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_CONNECTION: l2tp
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_INTERFACE: eth0
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_REQID: 29321
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_PROTO: esp
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_UNIQUEID: 41649
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_ME: 172.16.31.48
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_MY_ID: 172.16.31.48
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_MY_CLIENT: 172.16.31.48/32
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_MY_SOURCEIP: 
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_MY_PROTOCOL: 17
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_MY_PORT: 1701
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_PEER: 10.254.135.26
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_PEER_ID: 192.168.3.13
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_PEER_CLIENT: 10.254.135.26/32
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_PEER_SOURCEIP: 
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_PEER_PROTOCOL: 17
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_PEER_PORT: 1701
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_XAUTH_ID: 
Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CHD] updown:         PLUTO_UDP_ENC: 4500

why does this happen?it seems like the reqid 29228 still established.But the new reqid get the old connections's ip

Oct 19 09:07:15 13-59-shanghai-hw-cug charon: 15[CFG] unable to install policy 172.16.31.48/32[udp/l2tp] === 10.254.135.26/32[udp/l2tp] out for reqid 29321, the same policy for reqid 29228 exists

**Comment by Thermi** [comment]
the policies can't distinguish between hosts that use the same ports and IP addresses. This is a common problem with L2TP over IPsec with such hosts. for any firewall, the 5-tuples of a packet must be unique and what you show here is that you have two connections that have identical ones which can not be handled by a firewall. What are you even trying to do here? It makes no sense to establish two CHILD_SAs in transport mode for the same IP address because they cover the same traffic.
